.github/workflows/tests.yml

```diff
@@ -0,0 +1,277 @@
+name: Tests
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ dev ]
+
+jobs:
+
+  phpunit:
+
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+         image: postgres:9.6-alpine
+         options: --name=postgres
+         env:
+           POSTGRES_DB: testing
+           POSTGRES_USER: talentcloud
+           POSTGRES_PASSWORD: talentcloud
+         ports:
+          - 5432:5432
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: shivammathur/setup-php@v2
+      with:
+        php-version: 7.2
+        extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, bcmath, soap, intl, gd, exif, iconv, imagick
+        tools: composer:v1
+
+    - name: Copy .env
+      run: cp .env.travis .env
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-php-
+
+    - name: Install dependencies
+      if: steps.composer-cache.outputs.cache-hit != 'true'
+      uses: MilesChou/composer-action/7.2/install@master
+      with:
+        args: install --no-interaction --prefer-dist --no-progress --no-suggest
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.6.0
+
+    - name: Install npm dependencies
+      if: steps.npm-cache.outputs.cache-hit != 'true'
+      run: npm install
+
+    - name: Cache Npm packages
+      id: npm-cache
+      uses: actions/cache@v2
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - name: Generate key
+      run: php artisan key:generate
+
+    - name: Directory Permissions
+      run: chmod -R 777 storage vendor bootstrap/cache
+
+    - name: Generate certificates
+      run: make gen-certs
+
+    - name: Create Laravel storage symlink
+      run: php artisan storage:link
+
+    - name: Setup database
+      run: |
+        docker exec -d postgres psql -c 'create database testing with owner talentcloud;' -U talentcloud --no-password
+        php artisan migrate --seed
+      env:
+        DB_PORT: 5432
+        DB_HOST: localhost
+        DB_DATABASE: testing
+
+    - name: Compile webpack assets
+      run: npm run prod
+
+    - name: Start a web server
+      run: php artisan serve &
```

Comment:
I'm not sure if this is necessary in the end. We should see if the tests can pass without it.